chain/client/src/stateless_validation/chunk_validator/orphan_witness_handling.rs

```diff
@@ -77,31 +76,8 @@ impl Client {
         let possible_epochs =
             self.epoch_manager.possible_epochs_of_height_around_tip(&chain_head, witness_height)?;
 
-        // Try to validate the witness assuming that it resides in one of the possible epochs.
-        // The witness must pass validation in one of these epochs before it can be admitted to the pool.
-        let mut epoch_validation_result: Option<Result<(), Error>> = None;
-        for epoch_id in possible_epochs {
-            match self.partially_validate_orphan_witness_in_epoch(&witness, &epoch_id) {
-                Ok(()) => {
-                    epoch_validation_result = Some(Ok(()));
-                    break;
-                }
-                Err(err) => epoch_validation_result = Some(Err(err)),
-            }
-        }
-        match epoch_validation_result {
-            Some(Ok(())) => {} // Validation passed in one of the possible epochs, witness can be added to the pool.
-            Some(Err(err)) => {
-                // Validation failed in all possible epochs, reject the witness
-                return Err(err);
-            }
-            None => {
-                // possible_epochs was empty. This shouldn't happen as all epochs around the chain head are known.
-                return Err(Error::Other(format!(
-                "Couldn't find any matching EpochId for orphan chunk state witness with height {}",
-                witness_height
-            )));
-            }
+        if !possible_epochs.contains(&witness.epoch_id) {
```

Comment:
Since we are doing `partially_validate_state_witness_in_epoch`, I think we can completely get rid of this check here. We can even delete the `possible_epochs_of_height_around_tip` function.

Reply:
I thought about that as well, but there is one nasty scenario where this could backfire: state witness has random non-existing `prev_block_hash` and some old epoch where the chunk producer happened to be a chunk producer for that height and shard id. While this is highly unlikely, I don't think we can just disregard this.
Later I'm planning adding more epoch and height checks to `partially_validate_state_witness_in_epoch` and then we can remove it, but let's keep it for now, this PR is big enough as it is right now.

Reply:
It's really nice that we don't have to validate in multiple epochs now, the `epoch_id`field is really useful :+1:.
But I feel that we'll have to keep `possible_epochs_of_height_around_tip` around, otherwise a malicious chunk producer might be able to spam the orphan witness pool.

Reply:
But don't we have `ALLOWED_ORPHAN_WITNESS_DISTANCE_FROM_HEAD.contains(&head_distance)` to handle such scenarios?

Reply:
- we don't have chunk to verify chunk header
- chunk_producer can not be faked as we are verifying signature
- we verify if chunk_producer is the correct chunk producer for the epoch and height (here it shouldn't be possible to have some old/bad epoch_id for the given height?)
- we verify that the height parameter isn't too far away from head height

Reply:
Also, we can consider moving `ALLOWED_ORPHAN_WITNESS_DISTANCE_FROM_HEAD.contains(&head_distance)` to `partially_validate_state_witness`?

Reply:
As for why it's good to use `possible_epochs_of_height_around_tip`, I wrote a longer comment here: https://github.com/near/nearcore/pull/10613#discussion_r1499175365

Reply:
Will go through the comments and would have to think harder. It's still not super clear to me why we need this but will spend some time :)

Thanks @jancionear for the context!